core/src/db/entity/notifier.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{prisma::notifier, CoreError};
+use serde::{Deserialize, Serialize};
+use specta::Type;
+use std::str::FromStr;
+use utoipa::ToSchema;
+
+#[derive(Serialize, Deserialize, ToSchema, Type)]
+pub struct Notifier {
+	id: i32,
+	// Note: This isn't really needed, we could rely on tags. However, in order to have at least one
+	// readable field in the DB (since the config is dumped to bytes) I left this in
+	#[serde(rename = "type")]
+	_type: NotifierType,
+	config: NotifierConfig,
+}
+
+#[derive(Serialize, Deserialize, ToSchema, Type)]
+#[serde(untagged)]
+pub enum NotifierConfig {
+	Discord {
+		webhook_url: String,
+	},
+	Telegram {
+		encrypted_token: String,
```

Comment:
If we are storing an encrypted token for Telegram, we'll have to accepted an unencrypted token value during the creation flow. No UI exists yet, so this can be a follow up 🙂